packages/botonic-pulumi/src/index.ts

```diff
@@ -2,10 +2,27 @@ import { Config } from '@pulumi/pulumi'
 import { join } from 'path'
 import { cwd } from 'process'
 
-export const getNamePrefix = (): string => {
+export const PROJECT_NAME_SEPARATOR = '-'
+export const MAX_PROJECT_NAME_LENGTH = 30
+
+export function getNamePrefix(): string {
   const config = new Config()
-  return `botonic-${config.get('projectName')}-${config.get('stackName')}`
+  return generatePrefix(
+    config.get('projectName') as string,
+    config.get('stackName') as string
+  )
+}
+
+export function generatePrefix(projectName: string, stackName: string): string {
```

Comment:
Better more explicit function name: `generateStackNamePrefix()`

Reply:
I have changed it to `generateProjectStackNamePrefix`, which is the most explicit one.